### PR TITLE
Dynamic columns as a function prop

### PR DESCRIPTION
**Pull request by levibuzolic**
Quick feature request, this makes it much easier/quicker to provide dynamic column counts without needing to measure the container in user-code. I think this is a nice interface and feels appropriate given `react-photo-gallery` is already doing the measurement.

I've bumped the minor version to indicate new functionality which should be non-breaking.

### CODE REVIEW

**Comment by levibuzolic** [comment]
Also removed the `package-lock.json` which is long long out-of-date and it seems the `yarn.lock` is more recent. 

Ideally projects should use either a `yarn.lock` or a `package-lock.json` not both -- as it leads to package inconsistencies.

For example if you currently try and `npm install` you'll end up with vastly different dependencies, resulting in tests failing.

File: __test__/Gallery.test.js [hunk 1]
```diff
@@ -20,37 +20,48 @@ describe('Gallery', () => {
...
     expect(wrapper).toMatchSnapshot();
   });
 
   it('renders correctly with direction set to column', () => {
-     wrapper = mount(
+    wrapper = mount(
       <Gallery photos={photos} onClick={handleClick} direction={'column'}/>
     );
-    wrapper.setState({containerWidth: '1139'})
+    wrapper.setState({containerWidth: 1139})
```
Comment (levibuzolic) [inline __test__/Gallery.test.js]: Noticed these `setStates` were setting the width as a string, which doesn't seem correct, so i've updated them.

**Comment by neptunian** [comment]
Great idea!  I was thinking instead of creating a new Code Sandbox example we can just adjust this already existing one to stop using the Measure component: https://codesandbox.io/s/ll7ym48027

File: README.md [hunk 2]
```diff
@@ -37,6 +37,7 @@ yarn add react-photo-gallery
 * [Custom Dynamic Columns](https://codesandbox.io/s/ll7ym48027)
 * [Selection using custom ImageComponent](https://codesandbox.io/s/o7o241q09)
 * [Sortable with drag and drop](https://codesandbox.io/s/8y7n1r9y5j)
+* [Dynamic Columns Function](https://codesandbox.io/s/o5mnq449)
```
Comment (neptunian) [inline README.md]: can you remove this?

File: docs/README.md [hunk 3]
```diff
@@ -41,6 +41,7 @@ yarn add react-photo-gallery
 * [Custom Dynamic Columns](https://codesandbox.io/s/ll7ym48027)
 * [Selection using custom ImageComponent](https://codesandbox.io/s/o7o241q09)
 * [Sortable with drag and drop](https://codesandbox.io/s/8y7n1r9y5j)
+* [Dynamic Columns Function](https://codesandbox.io/s/o5mnq449)
```
Comment (neptunian) [inline docs/README.md]: can you remove this?

**Comment by levibuzolic** [comment]
@neptunian code sandbox examples removed, thanks! :+1:

**Pull request merged**